src/Azure.Deployments.Extensibility.Providers.Graph/GraphProvider.cs

```diff
@@ -0,0 +1,403 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Deployments.Extensibility.Core;
+using Azure.Deployments.Extensibility.Core.Json;
+using Azure.Deployments.Extensibility.Core.Exceptions;
+using Azure.Deployments.Extensibility.Core.Validators;
+using Json.Pointer;
+using System.Net;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace Azure.Deployments.Extensibility.Providers.Graph
+{
+    public class GraphProvider : IExtensibilityProvider
+    {
+        public static readonly string ProviderName = "Graph";
+
+        private static readonly string InternalError = "InternalError";
+        private static readonly string DisplayName = "displayName";
+        private static readonly string GraphToken = "graphToken";
+        private static readonly string Name = "name"; // name is required in Bicep template for Graph extensible types
+        private static readonly string Id = "id";
+        private static readonly string Value = "value";
+        private static readonly IReadOnlyDictionary<string, string> AlternateKeyByType = new Dictionary<string, string>()
+        {
+            ["users"] = "userPrincipalName",
+        };
+
+        // This is a temporary solution for POC to recognize operations on entity collections
+        // TODO: Find a more consistent way and type generic way to handle the cases
+        //       Waiting on https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1997031
```

Comment:
Let's not refer to Microsoft internal tracking systems in comments or code -- these aren't useful to the community.